README.md
Java packaging scripts
======================

This is a no-install version of the scripts I maintain in Debian java-package package.
You may currently find it on Debian unstable (sid) or in the Debian git repository (git+ssh://git.debian.org/git/pkg-java/java-package.git)

The scripts create a Debian package from a Java(TM) 2 distribution.
It works currently with the following Java(TM) 2 REs, SDKs and APIs:
 Oracle(TM) 1.6 Standard Edition (starting with update 10)
 Oracle(TM) 1.7 Standard Edition


Usage
-----

> git clone git@github.com:cedricpineau/java-package.git
> cd java-package

Download a supported Java(TM) 2 RE or SDK or API tar.gz or self-extracting archive from http://www.oracle.com/technetwork/java/javase/downloads

> J2SE_PACKAGE_LIBDIR=./lib ./make-jpkg <downloaded-archive>


debian/changelog
java-package (0.50) UNRELEASED; urgency=low

  * Java-package reset for reintroduction in debian (removal of all
    previously supported jvm.
  * Add support for Oracle Java 7

 -- Cédric Pineau <cedric.pineau@gmail.com>  Thu, 22 Dec 2011 11:42:13 +0100

java-package (0.43) UNRELEASED; urgency=low

  * Remove Java plugins for all browsers (Closes: #367887).
  * Add support for Sun Java 6 Update XX (Closes: #514364).

 -- Michael Koch <konqueror@gmx.de>  Thu, 17 Sep 2009 13:15:55 +0200

java-package (0.42) unstable; urgency=low

  * Provide java*-runtime-headless and java*-sdk names for generated
    packages.
  * Recommend dpkg-dev and gcc. Closes: #281766, #405814.

 -- Matthias Klose <doko@debian.org>  Sat, 02 Aug 2008 12:10:52 +0200

java-package (0.41) unstable; urgency=low

  * Added support for new SUN JDK/JRE for amd64 downloads. Thanks to
    Artyom Gavrichenkov for the initial patch.

 -- Michael Koch <konqueror@gmx.de>  Fri, 21 Mar 2008 17:20:35 +0100

java-package (0.40) unstable; urgency=low

  * Add support for ibm-java2-sdk-5.0-6.0-linux-$arch.tgz. Closes: #462399
    Thanks to Evgeni Dobrev for the patch.
  * Fixed spelling error in long description: debian -> Debian.

 -- Michael Koch <konqueror@gmx.de>  Thu, 24 Jan 2008 22:48:46 +0100

java-package (0.39) unstable; urgency=low

  * Added support for IBM JDK 6.0 final release.
  * Updated Standard-Version to 3.7.3.
  * Updated debhelper level to 5.

 -- Michael Koch <konqueror@gmx.de>  Tue, 18 Dec 2007 07:38:34 +0100

java-package (0.38) unstable; urgency=low

  * Mention SUN JDK 6 in REAME.Debian. Closes: #448258.
  * Fixed typo when creating version number for SUN JDK 6. Closes: #451935.
  * Fix handling of symlinks for ControlPanel, HtmlConverter and java-rmi.cgi
    from SUN JDK 6. Closes: #448351.

 -- Michael Koch <konqueror@gmx.de>  Fri, 30 Nov 2007 00:03:50 +0100

java-package (0.37) unstable; urgency=low

  * Added libstdc++5 to Recommends of generated package. Closes: #446334.

 -- Michael Koch <konqueror@gmx.de>  Thu, 25 Oct 2007 22:45:42 +0200

java-package (0.36) unstable; urgency=low

  * Fixed package description to include more supported JDKs.
    Closes: #447615, #447617.

 -- Michael Koch <konqueror@gmx.de>  Wed, 24 Oct 2007 09:56:52 +0200

java-package (0.35) unstable; urgency=low

  * Use 'date -R' instead of the deprecated '822-date'. Closes: #441229.
  * Fix generation of .changes file. Closes: #332643.
    Thanks to Christoph Goehre for the patch.
  * Add possibility to add a configured revision to the version.
    Closes: #442246. Thanks to Christoph Goehre again for the patch.

 -- Michael Koch <konqueror@gmx.de>  Fri, 14 Sep 2007 17:28:13 +0200

java-package (0.34) unstable; urgency=low

  * Fixed SUN JDK on amd64. Closes: #439221.

 -- Michael Koch <konqueror@gmx.de>  Fri, 24 Aug 2007 00:41:54 +0200

java-package (0.33) unstable; urgency=low

  * Fixed postinst scripts for SUN JDK 1.4, 5 and 6. Closes: #435632.

 -- Michael Koch <konqueror@gmx.de>  Thu, 02 Aug 2007 18:43:40 +0200

java-package (0.32) unstable; urgency=low

  * Added support for iceweasel. Closes: #433445.
  * Fix dangling links for SUN Java 5. Closes: #305053.
  * Depend on libx11-6 and libasound2. Closes: #289004.
  * Fix building on ia64. Closes: #355465.

 -- Michael Koch <konqueror@gmx.de>  Thu, 19 Jul 2007 22:51:31 +0200

java-package (0.31) unstable; urgency=low

  * Reapplied changes from 0.28 to 0.29. Closes: 322843.

 -- Michael Koch <konqueror@gmx.de>  Tue, 10 Jul 2007 08:16:14 +0200

java-package (0.30) unstable; urgency=low

  * Added support for SUN JDK 6u2 (and following updates).
  * Added support for ibm-java2-jre-5.0-5.0-linux-ppc.tgz.
    Thanks to B. Zhang.

 -- Michael Koch <konqueror@gmx.de>  Sun, 08 Jul 2007 09:33:29 +0200

java-package (0.29) unstable; urgency=low

  * Added support for SUN Java 6 and IBM Java 6.
    Closes: #322843, #402642, #418592, #425508, #430026.
  * Added myself to Uploaders.

 -- Michael Koch <konqueror@gmx.de>  Tue, 26 Jun 2007 17:40:04 -0100

java-package (0.28) unstable; urgency=low

  * Modified to reflect migration to Subversion
  * Bumped standards to version 3.7.2, no changes
  * Corrected test for AMD64 architecture during plug-in detection
  * Corrected detected version for IBM Java 5 JDK on AMD64 (Closes: #345775)
  * Repaired alternatives handling for IBM Java 5 JREs and JDKs
    (Closes: #338569)
  * Corrected pathing for alternatives installation for Mozilla and
    Firefox plugins (Closes: #341751)
  * Relaxed architectural restrictions on plugin alternative 
    installation for IBM Java 5 JREs and JDKs (Closes: #343837)
  * Added plugin installation for the new packaging of Firefox 1.5 and
    later (Closes: #341749)
  * Modified warning suppression for dh_shlibdeps to accommodate 
    changes in dpkg-dev (Closes: #354427)
  * Added quotes to variable tests for maintainer name and email 
    (Closes: #342623)
  * Refactored architecture evaluation to eliminate repetition and
    evaluate DEB_BUILD_GNU_TYPE correctly for i486-linux-gnu
  * Removed dependency on 'coreutils | stat' from control file, as
    the dependency is satisfied within essential (Closes: #370002)

 -- Barry Hawkins <barry@alltc.com>  Mon, 26 Jun 2006 11:19:08 -0400

java-package (0.27) unstable; urgency=low

  * Added support for IBM Java2 5.0 JREs and JDKs (Closes: #337129)
  * Corrected missing alternatives links for all JREs and JDKs
    (Closes: #317492)
  * Added conditional wrapping around dh_compress to account for
    JRE/JDKs that do not ship manpages (Closes: 301738)
  * Removed java-common from Provides: for JRE and JDK packages.

 -- Barry Hawkins <barry@bytemason.org>  Tue, 08 Nov 2005 12:16:24 -0500

java-package (0.26) unstable; urgency=low

  * Update standards version to 3.6.2
  * Resolve issues with powerpc-linux support detection (Closes: #317671)
  * Added support for Blackdown 1.4.2+02 release (Closes: #316831)

lib/ibm-j2re.sh
# Detect product
function ibm_j2re_detect() {
  local found=
  
  case "${DEB_BUILD_ARCH:-$DEB_BUILD_GNU_TYPE}" in
    i386|i486-linux-gnu)
      case "$archive_name" in
        "IBMJava2-JRE-131.tgz") # SUPPORTED
	    j2se_version=1.3.1${revision}
	    j2se_expected_min_size=20 #  bytes
	    found=true
	    ;;
        "IBMJava2-JRE-142.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=37 #  bytes
	    found=true
	    ;;
        "ibm-java2-jre-50-linux-i386.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=40 # 41655352 bytes
	    found=true
	    ;;
      esac
      ;;
    amd64|x86_64-linux-gnu)
      case "$archive_name" in
        "IBMJava2-JRE-AMD64-142.x86_64.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=37
	    found=true
	    ;;
        "ibm-java2-jre-50-linux-x86_64.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=46 # 48018851 bytes
	    found=true
	    ;;
      esac
      ;;
    powerpc|powerpc-linux)
      case "$archive_name" in
        "IBMJava2-JRE-131.ppc.tgz") # SUPPORTED
	    j2se_version=1.3.1${revision}
	    j2se_expected_min_size=20 #  bytes
	    found=true
	    ;;
        "IBMJava2-JRE-142.ppc.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=40 #  bytes
	    found=true
	    ;;
        "ibm-java2-jre-50-linux-ppc.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=42 # 43687702 bytes
	    found=true
	    ;;
	"ibm-java2-jre-5.0-5.0-linux-ppc.tgz") # SUPPORTED
            j2se_version=1.5.0${revision}
            j2se_expected_min_size=50 # 50066027 bytes
            found=true
            ;;
      esac
      ;;
    powerpc64|powerpc64-linux)
      case "$archive_name" in
        "IBMJava2-JRE-142.ppc64.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=37
	    found=true
	    ;;
        "ibm-java2-jre-50-linux-ppc64.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=46 # 47734431 bytes
	    found=true
	    ;;
      esac
      ;;
    s390|s390-linux)
      case "$archive_name" in
        "IBMJava2-JRE-131.s390.tgz") # SUPPORTED
	    j2se_version=1.3.1${revision}
	    j2se_expected_min_size=20
	    found=true
	    ;;
        "IBMJava2-JRE-141.s390.tgz") # SUPPORTED
	    j2se_version=1.4.1${revision}
	    j2se_expected_min_size=31
	    found=true
	    ;;
        "IBMJava2-JRE-142.s390.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=34
	    found=true
	    ;;
        "IBMJava2-JRE-142.s390x.tgz") # SUPPORTED
	    j2se_version=1.4.2${revision}
	    j2se_expected_min_size=34
	    found=true
	    ;;
        "ibm-java2-jre-50-s390.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=42 # 43686815 bytes
	    found=true
	    ;;
        "ibm-java2-jre-50-s390x.tgz") # SUPPORTED
	    j2se_version=1.5.0${revision}
	    j2se_expected_min_size=42 # 43773066 bytes
	    found=true
	    ;;
      esac
      ;;
  esac
  if [[ -n "$found" ]]; then
    cat << EOF

Detected product:
    Java(TM) Runtime Environment (JRE)
    Standard Edition, Version $j2se_version
    IBM Corporation
EOF
	if read_yn "Is this correct [Y/n]: "; then
	    j2se_found=true
	    j2se_release="${j2se_version:0:3}"
	    j2se_required_space=$(( $j2se_expected_min_size * 2 + 20 ))
	    j2se_vendor="ibm"
	    j2se_title="Java(TM) JRE, Standard Edition, IBM Corporation"
	    j2re_run
 	fi
    fi
}

j2se_detect_ibm_j2re=ibm_j2re_detect
